test/testdrive/linked-clusters.td

```diff
@@ -0,0 +1,305 @@
+> CREATE SINK snk2 FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM WITH (SIZE = '2')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_snk1
+materialize_public_snk2
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                  r1      ${arg.default-replica-size}
+mz_introspection         r1      1
+mz_system                r1      1
+materialize_public_snk1  linked  1
+materialize_public_snk2  linked  2
+
+# Test that changing a sink size results in the cluster being dropped and
+# recreated.
+> ALTER SINK snk2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  4
+
+# Test that a sink can be converted to a REMOTE sink and then back to a managed
+# sink. (Unsafe mode only.)
+> ALTER SINK snk2 SET (REMOTE 'localhost:32323')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+> ALTER SINK snk2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  4
+> ALTER SINK snk2 RESET (SIZE)
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  ${arg.default-storage-size}
+
+# Test that only the default clusters and replicas remain after dropping
+# the sinks.
+> DROP SINK snk1
+> DROP SINK snk2
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+
+# Test that two objects with the same name create distinct clusters
+# as the schema name is included in the cluster name
+
+> CREATE SCHEMA schema1;
+> CREATE SCHEMA schema2;
+
+> CREATE SOURCE schema1.s1 FROM LOAD GENERATOR COUNTER
+> CREATE SINK schema2.s1 FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM
+
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_schema1_s1
+materialize_schema2_s1
+
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
+materialize_schema1_s1  linked  ${arg.default-storage-size}
+materialize_schema2_s1  linked  ${arg.default-storage-size}
+
+> DROP SCHEMA schema1 CASCADE;
+> DROP SCHEMA schema2 CASCADE;
+
+# Test capitalizarion behavior
+
+> CREATE SOURCE "s1" FROM LOAD GENERATOR COUNTER
+> CREATE SINK "S1" FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM
+
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_s1
+materialize_public_S1
```

Comment:
Is this the correct behavior? I thought that generally identifiers are case-insensitve. For example:
```
materialize=> CREATE CLUSTER a REPLICAS (linked (SIZE '1'));
CREATE CLUSTER
materialize=> CREATE CLUSTER A REPLICAS (linked (SIZE '1'));
ERROR:  cluster 'a' already exists
materialize=> CREATE MATERIALIZED VIEW v IN CLUSTER A AS SELECT 1;
CREATE MATERIALIZED VIEW
```

Reply:
Ahh, I guess it's the quotes that make the difference.

Reply:
```
materialize=> CREATE CLUSTER a REPLICAS (linked (SIZE '1'));
CREATE CLUSTER
materialize=> CREATE CLUSTER "A" REPLICAS (linked (SIZE '1'));
CREATE CLUSTER
materialize=> SHOW CLUSTERS;
       name       
------------------
 A
 a
 default
 mz_introspection
 mz_system
(5 rows)
```